web/submission/views.py

```diff
@@ -55,22 +57,164 @@
 logger = logging.getLogger(__name__)
 
 
-def get_form_data():
+def parse_expr(expr, context):
+    """Return the value from a python AST expression.
+
+    Recursive! the initial call is the right hand side of an assignment.
+    Recursion is necessary because the expression is made up of a variable number
+    of subexpressions, sub-subexpressions, etc.
+    """
```

Comment:
In order to display the helpful package information in the web UI, we have to read & parse the text of each python file in the windows (& linux) `packages` directories.

This includes being able to understand a python `f` string, replacing tokens (such as `OPT_ARGUMENTS`) with the values they represent.